tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/SdkBuildTool.cs

```diff
@@ -0,0 +1,199 @@
+        }
+
+        public override Command GetCommand()
+        {
+            var command = new Command(BuildSdkCommandName, "Builds SDK source code for a specified language and project.");
+            command.SetHandler(async ctx => { await HandleCommand(ctx, ctx.GetCancellationToken()); });
+            command.AddOption(SharedOptions.PackagePath);
+
+            return command;
+        }
+
+        public async override Task HandleCommand(InvocationContext ctx, CancellationToken ct)
+        {
+            var command = ctx.ParseResult.CommandResult.Command.Name;
+            var commandParser = ctx.ParseResult;
+            var packagePath = commandParser.GetValueForOption(SharedOptions.PackagePath);
+            var buildResult = await BuildSdkAsync(packagePath, ct);
+            ctx.ExitCode = ExitCode;
+            _output.Output(buildResult);
+        }
+
+        [McpServerTool(Name = "azsdk_package_build_code"), Description("Build/compile SDK code for a specified project locally.")]
+        public async Task<DefaultCommandResponse> BuildSdkAsync(
+            [Description("Absolute path to the SDK project.")]
+            string packagePath,
+            CancellationToken ct = default)
+        {
+            try
+            {
+                _logger.LogInformation($"Building SDK for project path: {packagePath}");
+
+                // Validate inputs
+                if (string.IsNullOrEmpty(packagePath))
+                {
+                    return CreateFailureResponse("Package path is required.");
+                }
+
+                if (!Directory.Exists(packagePath))
+                {
+                    return CreateFailureResponse($"Path does not exist: {packagePath}");
+                }
+
+                // Get repository root path from project path
+                string sdkRepoRoot = _gitHelper.DiscoverRepoRoot(packagePath);
+                if (string.IsNullOrEmpty(sdkRepoRoot))
+                {
+                    return CreateFailureResponse($"Failed to discover local sdk repo with project-path: {packagePath}.");
+                }
+
+                _logger.LogInformation($"Repository root path: {sdkRepoRoot}");
+
+                string sdkRepoName = _gitHelper.GetRepoName(sdkRepoRoot);
+                _logger.LogInformation($"Repository name: {sdkRepoName}");
+
+                // Return if the project is python project
+                if (sdkRepoName.Contains(AzureSdkForPythonRepoName, StringComparison.OrdinalIgnoreCase))
+                {
+                    _logger.LogInformation("Python SDK project detected. Skipping build step as Python SDKs do not require a build process.");
+                    return CreateSuccessResponse("Python SDK project detected. Skipping build step as Python SDKs do not require a build process.");
+                }
+
+                // Get the build configuration (command or script path)
+                ProcessOptions options;
+                try
```

Comment:
nit: could this try block be refactored into a private method `CreateProcessOptions`?

Reply:
Sounds good to me.